linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -3263,7 +3269,11 @@ private boolean advance() {
         TKey leftKey2 = outerKeySelector.apply(left);
         int c = leftKey.compareTo(leftKey2);
         if (c != 0) {
-          assert c < 0 : "not sorted";
+          if (c > 0) {
+            throw new IllegalStateException(
+              "mergeJoin assumes sorted in ascending order input, "
+                 + "however " + leftKey + " is less than " + leftKey2);
```

Comment:
`leftKey + " is less than " + leftKey2` seems to be wrong.

`c = leftKey.compareTo(leftKey2)`.
`c > 0 ` ==> `leftKey > leftKey2`

Reply:
yes, sure, I wrote the comment when the assertion was wrong :-)
sorry